### DISCUSSION TITLE
Trouble organizing into scenes

### DISCUSSION THREAD

**Discussion by oaklandgit** [replit/kaboom#203] [Q&A] [answered]
Hi all,

I'm stuck on a bug with my code and I'm not sure if it's my coding error or something with Kaboom. But maybe there's something I'm not understanding about scope within scenes.

In version A I am not enclosing anything in a scene scope and everything works (see the script.js file)
https://replit.com/@LarryStone/A-no-scenes?v=1

In version B, I am enclosing my functions in a scene scope and then calling the go function, and I get a mysterious "script error"
https://replit.com/@LarryStone/B-scenes?v=1

Thank you in advance for any nudges in the right direction!

**Comment by oaklandgit** [comment, marked as answer]
OK I figured out my issue, but it raises a question about how Kaboom should behave…

The problem was that loading resources (spritesheets for example) _from within the scene scope_ leaves Kaboom unable to find those resources. But I expected those resources to be available at least from within the scene scope.

**Reply by slmjkdbtl** [reply]
Yeah the loader doesn't work mid-game now. But I think this should work:
```js
Promise.all([
     loadSprite("sprite1", ...),
     loadSprite("sprite2", ...),
]).then(() => {
    add([
         sprite("sprite2"),
    ]);
})
```

**Reply by slmjkdbtl** [reply]
I wonder why do you have to load assets in scene scope, instead of global? Do you have a lot of assets and have memory / load time concern?

**Reply by oaklandgit** [reply]
No, it's not a memory concern. It's about organizing the code -- I'd like to keep the resource loaders with the game objects they are for. 

But I'm ok loading then in the global scope.

**Reply by slmjkdbtl** [reply]
I am planning to support in-game loading assets tho

**Reply by oaklandgit** [reply]
Awesome!
